python/lsst/analysis/tools/tasks/catalogMatch.py

```diff
@@ -0,0 +1,408 @@
+        """Prep the catalog and run the matcher.
+
+        Parameters
+        ----------
+        catalog : `pandas.core.frame.DataFrame`
+
+        Returns
+        -------
+        `pipeBase.Struct` containing:
+            matchedCat : `pandas.core.frame.DataFrame`
+                Catalog containing the matched objects with all columns from
+                the original input catalogs, with the suffix "_ref" or
+                "_target" for duplicated column names, plus a column with the
+                angular separation in arcseconds between matches.
+        """
+        # Apply the selectors to the catalog
+        mask = np.ones(len(catalog), dtype=bool)
+        for selector in self.config.selectorActions:
+            mask &= selector(catalog)
+
+        for selector in self.config.sourceSelectorActions:
+            mask &= selector(catalog, band=self.config.selectorBand).astype(bool)
+
+        targetCatalog = catalog[mask]
+        targetCatalog = targetCatalog.reset_index()
+
+        if (len(targetCatalog) == 0) or (len(self.refCat) == 0):
+            matches = pipeBase.Struct(refMatchIndices=np.array([]),
+                                      targetMatchIndices=np.array([]),
+                                      separations=np.array([]))
+        else:
+            # Run the matcher
+            matches = self.matcher.run(self.refCat, targetCatalog)
+
+        # Join the catalogs for the matched catalogs
+        refMatches = self.refCat.iloc[matches.refMatchIndices].reset_index()
+        sourceMatches = targetCatalog.iloc[matches.targetMatchIndices].reset_index()
+        matchedCat = sourceMatches.join(refMatches, lsuffix='_target', rsuffix='_ref')
+
+        separations = pd.Series(matches.separations).rename("separation")
+        matchedCat = matchedCat.join(separations)
+
+        return pipeBase.Struct(matchCatalog=matchedCat)
+
+    def setRefCat(self, skymap, tract):
+        """Make a reference catalog with coordinates in degrees
+
+        Parameters
+        ----------
+        skymap : `lsst.skymap`
+            The skymap used to define the patch boundaries.
+        tract : int
+            The tract corresponding to the catalog data.
+        """
+        # Load the reference objects in a skyCircle around the tract
+        tractInfo = skymap.generateTract(tract)
+        boundingCircle = tractInfo.getOuterSkyPolygon().getBoundingCircle()
+        center = lsst.geom.SpherePoint(boundingCircle.getCenter())
+        radius = boundingCircle.getOpeningAngle()
+
+        epoch = Time(self.config.epoch, format="decimalyear")
+
+        skyCircle = self.refObjLoader.loadSkyCircle(center,
+                                                    radius,
+                                                    'i',
+                                                    epoch=epoch)
+        refCat = skyCircle.refCat
+
+        # Convert the coordinates to RA/Dec and convert the catalog to a
+        # dataframe
```

Comment:
Spaces around *

Reply:
Are you using black, this is a thing that has a right answer, and black does it.

Reply:
Yeah, when I tried to commit, triggering `pre-commit`, `black` complained about all the places I just removed whitespaces. However, this seems to disagree with https://developer.lsst.io/python/style.html#binary-operators-should-be-surrounded-by-a-single-space-except-for. Maybe I don't have `black` configured correctly.

Reply:
When black is enabled for a project the Python style guide for formatting is no longer relevant. I thought there was a comment about that somewhere in the dev guide but maybe that was only in the section about using black.

Reply:
https://developer.lsst.io/python/style.html?highlight=black#black-may-be-used-to-fix-pep-8-compliance